### ISSUE TITLE
[New Feature] Add Schema Diagram Panel

### ISSUE THREAD

**Issue by sky5walk** [sqlitebrowser/sqlitebrowser#1260]
### Details for the issue
An informative feature I miss from other commercial browsers is a graphical representation of the underlying db schema. While they often support input mechanisms to create tables and fields, I would be happy with just a movable/resizable static representation of the current db. This would allow printing and sharing concepts in the db design and evolution with end users.

### Useful extra information
I'm sure the DB4S developers would have a blast coding up the graphical view of the loaded db. :)
And it could expand over time to include input mechanisms that update/modify the db.
#### I'm opening this issue because:

- [ ] DB4S is crashing
- [ ] DB4S has a bug
- [x] DB4S needs a feature
- [ ] DB4S has another problem

#### I'm using DB4S on:

- [x] Windows: ( _10 Pro___ )
- [ ] Linux: ( _distro:_ ___ )
- [ ] Mac OS: ( _version:_ ___ )
- [ ] Other: ___

#### I'm using DB4S version:

- [ ] 3.10.1
- [ ] 3.10.0
- [ ] 3.9.1
- [x] Other: ___3.10.99 build 171205

#### I have also:

- [x] Tried out the latest nightly version: https://github.com/sqlitebrowser/sqlitebrowser#nightly-builds
- [x] Searched for an existing similar issue: https://github.com/sqlitebrowser/sqlitebrowser/issues?utf8=%E2%9C%93&q=is%3Aissue%20

**Comment by justinclift** [comment]
Yeah, having some kind of entity relationship diagramming has been talked about before.  It would be a useful new capability. :smile:

**Comment by mgrojo** [comment]
I saw this one day: #913. Where are those changes supposed to be?

**Comment by justinclift** [comment]
Ahhh yeah, that's one of the previous items about this. :smile:

@schdub created an initial rough Entity Relationship (ER) diagram thing for DB4S a while ago, but never got around to making a PR from it.  I saw some screenshots from it, and it seemed like a decent initial implementation that people would be able to use decently.

@innermous wanted to do something like that too, but didn't like the look of @schdub's approach so ignored it and didn't get very far.

As a thought, I'd recommend seeing if @schdub still has his code around.  He's a pretty good C++ developer, so it's likely to be pretty decent code in itself.  It might not take huge effort to get working with our current code base either. :smile:

**Comment by sky5walk** [comment]
#1962 looks great. When will this be available in the nightlies?

**Comment by mgrojo** [comment]
> #1962 looks great. When will this be available in the nightlies?

It still needs a lot of work. The diagram cannot be saved, nor printed. The relations are simple lines between tables; they don't show which table is the parent and which the child, nor which columns are the keys involved in the relation. Among other pending features listed in #1962. I guess it won't make to the next stable release.
